packages/constellation/src/components/Compositions/Navigation/Navigation.types.ts

```diff
@@ -0,0 +1,111 @@
+import { FC } from 'react'
+
+import { ProfileProps } from '../../Primitives/Profile/Profile.types'
+
+interface Button {
+  onClick: () => void
+  text: string
+  variant?: 'primary' | 'secondary' | 'minimal'
+}
+
+interface tab {
+  /**
+   * displays a notification badge on the tab
+   */
+  dot?: boolean
+  /**
+   * the key of the tag, this is used to identify the currently selected tab.
+   */
+  key: string
+  /**
+   * The display name of the tab
+   */
+  label: string
```

Comment:
Since we translate all of our copy on the site, anywhere there will be text displayed to the user should take a react component as the type so we can pass our <FormattedMessage /> component into the nav component.